packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster-private-endpoint-subnet-filter.ts

```diff
@@ -0,0 +1,59 @@
+/// !cdk-integ pragma:disable-update-workflow
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import { App } from 'aws-cdk-lib';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import * as eks from 'aws-cdk-lib/aws-eks';
+import { EC2_RESTRICT_DEFAULT_SECURITY_GROUP } from 'aws-cdk-lib/cx-api';
+import { getClusterVersionConfig } from './integ-tests-kubernetes-version';
+
+const app = new App({
+  postCliContext: {
+    '@aws-cdk/aws-lambda:createNewPoliciesWithAddToRolePolicy': false,
+  },
+});
+
+const stack = new cdk.Stack(app, 'StackWithVpc', {
+  env: {
+    region: 'eu-west-1',
+    account: '123456',
```

Comment:
Can you explain how this successfully deploys? sorry I might just not be familiar with the magic that replaces/overrides this at deploy time, and why is this needed?